### PR TITLE
Soback 31 port harvest calendar

### PR DESCRIPTION
**Pull request by jorntx**
Tricky PR and I don't expect you to review all in detail - good place to start is to ignore all .js files and concentrate on the .vue files.

Please note:

The playback/harvest calendar functionality will only work in full and all domains if you deploy the app to Tomcat and use it from Tomcat. You can test the calendar on you dev. server på using http://localhost:8081/calendar?url=http://www.ekot.dk

### CODE REVIEW

**Comment by jorntx** [comment]
Forgot to mention that a slipped in the title tag ticket in this PR...

File: src/js/public/solrwayback_index_page.html [hunk 1]
```diff
@@ -5,11 +5,11 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <link rel="icon" href="<%= BASE_URL %>favicon.ico">
-    <title><%= htmlWebpackPlugin.options.title %></title>
```
Comment (jesperlauridsen) [inline src/js/public/solrwayback_index_page.html:8]: This seems like the easy solution... no way to change the variable without going through the woods and back?

File: src/js/public/solrwayback_index_page.html [hunk 2]
```diff
@@ -5,11 +5,11 @@
...
+    <title>SolrWayback</title>
   </head>
   <body>
     <noscript>
-      <strong>We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
+      <strong>We're sorry but SolrWayback doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
```
Comment (jesperlauridsen) [inline src/js/public/solrwayback_index_page.html:12]: Same question :D

File: src/js/src/components/harvestCalendar/AllYearsGraph.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  </div>
+</template>
+
+<script>
+
+import {toHumanDate} from './util'
+
+export default {
+  name: 'AllYearsGraph',
+  components: {
```
Comment (jesperlauridsen) [inline src/js/src/components/harvestCalendar/AllYearsGraph.vue:30]: No components used, so this seems unnecessary.

File: src/js/src/components/harvestCalendar/plugins/weekdayActivityLevel.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Plugin to add the property activityLevel to every month in the data object.
+ * Takes a transformation function to choose the way the activity level is calculated.
+ * 
+ * @param {Object} datesObject The final object of years, months and days to add data to.
+ * @param {Function} transformationFunction The function to calculate activity level from transformation-functions.js
+ */
+export function addActivityLevelToWeeks(datesObject, transformationFunction) {
+
+    let {maximumHarvests} = getMaximumHarvestCount(datesObject)
```
Comment (jesperlauridsen) [inline src/js/src/components/harvestCalendar/plugins/weekdayActivityLevel.js:12]: What's gained by having the variable in {} here? It confuse me little bit.

File: src/js/src/views/HarvestCalendar.vue [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+  
+  
+  mounted () {
+    if (this.$route.query.url){
+       this.currentHarvestUrl = this.$route.query.url
+     }
+     requestService.getHarvestDates(encodeURIComponent(this.$route.query.url))
+        .then(data => {
+            if (data.dates === undefined || data.dates.length === 0) {
```
Comment (jesperlauridsen) [inline src/js/src/views/HarvestCalendar.vue:124]: Is this double check necessary? Im just curious, I don't know the returned data...

**Comment by jesperlauridsen** [comment]
This seems to work very nice when deployed. One question tho - when i navigate to a certain date where i have a harvest, and click the small github-layout square, nothing happens. Shouldn't something happen? Either a series of links of the different harvests of that date - or if theres only one, a new page with the requested site on the harvest time (or opened in the same tab)?

File: src/js/public/solrwayback_index_page.html (hunk 1 above)
Comment (jorntx) [reply src/js/public/solrwayback_index_page.html:8]: This variable is overwritten on runtime (e.g. if you step into the about page or just /solrwayback). So I choose not to use the htmlWebpackPlugin at all and replaced it with out generic title. Other than that I have no explanation for what I did :)

File: src/js/public/solrwayback_index_page.html (hunk 2 above)
Comment (jorntx) [reply src/js/public/solrwayback_index_page.html:12]: See above

File: src/js/src/components/harvestCalendar/AllYearsGraph.vue (hunk 3 above)
Comment (jorntx) [reply src/js/src/components/harvestCalendar/AllYearsGraph.vue:30]: Good catch - thanks!

File: src/js/src/views/HarvestCalendar.vue (hunk 5 above)
Comment (jorntx) [reply src/js/src/views/HarvestCalendar.vue:124]: No - I double checked (no pun intended :) )  only length check is needed. Good catch.

**Comment by jorntx** [comment]
Answer to questions:

1. Yes there should be links in the bottom....??
2. We always open in new tab.

File: src/js/public/solrwayback_index_page.html (hunk 1 above)
Comment (jesperlauridsen) [reply src/js/public/solrwayback_index_page.html:8]: :+1:

File: src/js/public/solrwayback_index_page.html (hunk 2 above)
Comment (jesperlauridsen) [reply src/js/public/solrwayback_index_page.html:12]: :+1:

**Comment by jesperlauridsen** [comment]
... I have no idea how i missed the links that appear when you click a harvest time. >.< sorry about that!

**Review by jesperlauridsen — APPROVED** [review]
very nice! Nice that we could reuse so much of it. Excellent work!

**Pull request merged**